test/unit/local/watcher.js

```diff
@@ -50,6 +51,50 @@ describe('LocalWatcher Tests', function () {
       this.watcher.start()
     })
 
+    it('only recomputes checksums of changed files', async function () {
+      const builders = new MetadataBuilders(this.pouch)
+      const unchangedFilename = 'unchanged-file.txt'
+      const changedFilename = 'changed-file.txt'
+      const unchangedPath = path.join(this.syncPath, unchangedFilename)
+      const changedPath = path.join(this.syncPath, changedFilename)
+      const unchangedData = 'Unchanged file content'
+      const changedData = 'Changed file initial content'
+      await fs.outputFile(unchangedPath, unchangedData)
+      await fs.outputFile(changedPath, changedData)
+      const unchangedStats = await fs.stat(unchangedPath)
+      const {ino: changedIno} = await fs.stat(changedPath)
+      const unchangedDoc = await builders.file()
+        .upToDate()
+        .path(unchangedFilename)
+        .data(unchangedData)
+        .stats(unchangedStats)
+        .create()
+      const changedDoc = await builders.file()
+        .upToDate()
+        .path(changedFilename)
+        .data(changedData)
+        .ino(changedIno)
+        .updatedAt(new Date('2017-03-19T16:44:39.102Z'))
+        .create()
+
+      await fs.outputFile(changedPath, 'Changed file NEW content')
+      this.prep.addFileAsync = sinon.stub().resolves()
+      sinon.spy(this.watcher, 'checksum')
+
+      try {
+        await this.watcher.start()
+
+        should(this.watcher.checksum.args).deepEqual([[changedFilename]])
```

Comment:
[nitpick-style] how about 
```
checksumer = sinon.spy(this.watcher, 'checksum')
should(checksumer.calledOnce && checksumer.calledWith(changedFilename))
````
 ?

Reply:
Isn't current version actually shorter? Although it would definitely make sense for more than 1 assertion...

Reply:
Well I agree your solution is actually more readable...